hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -139,22 +212,40 @@ static StorageDescriptor storageDescriptor(TableMetadata metadata, boolean hiveE
       storageDescriptor.setInputFormat("org.apache.hadoop.mapred.FileInputFormat");
       serDeInfo.setSerializationLib("org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe");
     }
+
     storageDescriptor.setSerdeInfo(serDeInfo);
     return storageDescriptor;
   }
 
-  static void cleanupMetadata(FileIO io, String commitStatus, String metadataLocation) {
+  default void cleanupMetadataAndUnlock(
```

Comment:
I would probably just leave `cleanupMetadata` as it is and introduce `cleanupMetadataAndUnlock()` as a separate method